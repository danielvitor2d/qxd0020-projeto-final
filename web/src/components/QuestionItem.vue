<script setup lang="ts">
defineProps<{
  item: {
    id: number
    label: string
    course: string
  }
  onDelete: (id: number) => void
}>()
</script>

<template>
  <div class="w-full flex flex-row items-center justify-between">
    <p class="flex-1">{{ item.label }}</p>
    <p class="mr-4">{{ item.course }}</p>
    <button @click="() => onDelete(item.id)">
      <font-awesome-icon icon="fa-xmark" />
    </button>
  </div>
</template>
