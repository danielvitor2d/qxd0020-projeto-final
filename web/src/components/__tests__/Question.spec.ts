import { mount } from '@vue/test-utils';
import { expect, describe, it, vi } from 'vitest';

import QuestionItem from '../../components/QuestionItem.vue';
import Question from '../../components/Question.vue';

describe('Question', () => {
  it('renders correctly', () => {
    const onDelete = vi.fn();
    const question = {
      id: 1,
      items: [
        { id: 1, label: 'Option 1', course: 'Course A' },
        { id: 2, label: 'Option 2', course: 'Course B' },
        { id: 3, label: 'Option 3', course: 'Course C' },
      ],
    };

    const wrapper = mount(Question, {
      props: {
        question,
        onDelete,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('calls onDelete when delete button is clicked', () => {
    const onDelete = vi.fn();
    const question = {
      id: 1,
      items: [
        { id: 1, label: 'Option 1', course: 'Course A' },
        { id: 2, label: 'Option 2', course: 'Course B' },
        { id: 3, label: 'Option 3', course: 'Course C' },
      ],
    };

    const wrapper = mount(Question, {
      props: {
        question,
        onDelete,
      },
    });

    const deleteButton = wrapper.find('.delete-button');
    deleteButton.trigger('click');

    expect(onDelete).toHaveBeenCalledWith(question.id);
  });

  it('adds a new item when add button is clicked', () => {
    const onDelete = vi.fn();
    const question = {
      id: 1,
      items: [
        { id: 1, label: 'Option 1', course: 'Course A' },
        { id: 2, label: 'Option 2', course: 'Course B' },
        { id: 3, label: 'Option 3', course: 'Course C' },
      ],
    };

    const wrapper = mount(Question, {
      props: {
        question,
        onDelete,
      },
    });

    const addButton = wrapper.find('.add-button');
    addButton.trigger('click');

    const items = wrapper.findAllComponents(QuestionItem);
    expect(items.length).toBe(3);
  });
});
