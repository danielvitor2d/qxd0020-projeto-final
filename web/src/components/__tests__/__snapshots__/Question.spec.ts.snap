// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Question > renders correctly 1`] = `
<div
  class="w-full h-fit p-5 bg-[#F5F5F5] border-[1px] border-gray-400 rounded-md"
>
  <div
    class="w-full mb-6 flex flex-row items-center justify-center gap-10"
  >
    <input
      class="w-fit flex-1 bg-[#F5F5F5] font-normal text-sm px-1 py-2 border-[1px] border-gray-400"
      type="text"
    />
    <button
      class="delete-button"
    >
      <font-awesome-icon
        class="w-fit text-red-600"
        icon="trash-can"
      />
    </button>
  </div>
  <div
    class="w-full flex flex-col gap-3"
  >
    
    <div
      class="w-full flex flex-row items-center justify-between"
    >
      <p
        class="flex-1"
      >
        Option 1
      </p>
      <p
        class="mr-4"
      >
        Course A
      </p>
      <button>
        <font-awesome-icon
          icon="fa-xmark"
        />
      </button>
    </div>
    <div
      class="w-full flex flex-row items-center justify-between"
    >
      <p
        class="flex-1"
      >
        Option 2
      </p>
      <p
        class="mr-4"
      >
        Course B
      </p>
      <button>
        <font-awesome-icon
          icon="fa-xmark"
        />
      </button>
    </div>
    <div
      class="w-full flex flex-row items-center justify-between"
    >
      <p
        class="flex-1"
      >
        Option 3
      </p>
      <p
        class="mr-4"
      >
        Course C
      </p>
      <button>
        <font-awesome-icon
          icon="fa-xmark"
        />
      </button>
    </div>
    
  </div>
  <button
    class="add-button flex flex-row gap-3 mt-3 px-4 py-2 items-center justify-center text-[#007AC2] rounded-md hover:bg-[#2ea6eb] hover:text-white"
  >
    <font-awesome-icon
      class="w-fit"
      icon="plus"
    />
     Adicionar opção 
  </button>
</div>
`;
